test/unit/response-types/code-response-type_test.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+/**
+ * Module dependencies.
+ */
+
+var CodeResponseType = require('../../../lib/response-types/code-response-type');
+var should = require('should');
+var sinon = require('sinon');
+
+/**
+ * Test `CodeResponseType`.
+ */
+
+describe('CodeResponseType', function() {
+  describe('generateAuthorizationCode()', function() {
+    it('should call `model.generateAuthorizationCode()`', function() {
+      var model = {
+        generateAuthorizationCode: sinon.stub().returns('21345'),
+        saveAuthorizationCode: function() {}
+      };
+      var handler = new CodeResponseType({ authorizationCodeLifetime: 120, model: model });
+
+      return handler.generateAuthorizationCode()
+        .then(function() {
+          model.generateAuthorizationCode.callCount.should.equal(1);
+        })
+        .catch(should.fail);
```

Comment:
As per your previous good point, we don't need these `.catch(should.fail);`.

Reply:
You missed the ones in the file I commented :laughing:

Reply:
Fixed now. 😅